lib/colorado_lottery.rb

```diff
@@ -0,0 +1,53 @@
+class ColoradoLottery
+  attr_reader :registered_contestants, :winners, :current_contestants
+
+  def initialize
+    @registered_contestants = {}
+    @winners = []
+    @current_contestants = {}
+  end
+
+  def interested_and_18?(contestant, game)
+    contestant.age >= 18 && contestant.game_interests.include?(game.name)
+  end
+
+  def out_of_state_only_for_national(contestant, game)
+    if game.national_drawing? == false
+      contestant.out_of_state? == false
+    else
+      true
+    end
+  end
+
+  def can_register?(contestant, game)
+    interested_and_18?(contestant, game) && out_of_state_only_for_national(contestant, game)
+  end
+
+  def register_contestant(contestant, game)
+    if can_register?(contestant, game) == true
+      if @registered_contestants[game.name] == nil
+        @registered_contestants[game.name] = [contestant]
+      elsif @registered_contestants[game.name] != nil
+      @registered_contestants[game.name] << contestant
+      end
+    end
+  end
+
+  def eligible_contestants(game)
+    @registered_contestants[game.name].find_all do |contestant|
+      contestant.spending_money >= game.cost
+    end
+  end
+
+  def charge_contestants(game)
+    @current_contestants[game] = []
+    eligible_contestants(game).each do |contestant|
+      contestant.spending_money -= game.cost
+      @current_contestants[game] << contestant.full_name
```

Comment:
Unfortunately line 46 will not run quite right here, which I imagine you experienced. For this to run, our current_contestants hash needs to be initialized in a way that gives it an array as a default value. So up above on line 7, for this to work it would need to read: 
`@current_contestants = Hash.new { |hash, key| hash[key] = [] }`